task-driver/src/helpers.rs

```diff
@@ -337,44 +335,39 @@ pub(crate) async fn update_wallet_validity_proofs(
     Ok(())
 }
 
-/// Attach a copy of the witness to the locally managed state
-///
-/// This witness is referenced by match computations to provide linkable
-/// commitments into a previously proven validity bundle
-async fn record_validity_witness(
-    order_id: &OrderIdentifier,
-    commitments_witness: &SizedValidCommitmentsWitness,
-    valid_reblind_witness: &SizedValidReblindWitness,
-    global_state: &RelayerState,
-) {
-    let witness_bundle = OrderValidityWitnessBundle {
-        reblind_witness: Arc::new(valid_reblind_witness.clone()),
-        commitment_witness: Arc::new(commitments_witness.clone()),
-    };
-
-    global_state
-        .read_order_book()
-        .await
-        .attach_validity_proof_witness(order_id, witness_bundle)
-        .await;
-}
-
-/// Attach a validity proof to the locally managed state
+/// Attach a validity proof and witness to the locally managed state
 ///
 /// This is gossipped to the network so that peers may verify the validity
 /// bundle and schedule matches on the proven order
-pub(crate) async fn record_validity_proof(
+async fn record_validity_proof_and_witness(
     order_id: &OrderIdentifier,
-    commitments_bundle: &ValidCommitmentsBundle,
-    reblind_bundle: &ValidReblindBundle,
+    commitments_witness: &SizedValidCommitmentsWitness,
+    reblind_witness: &SizedValidReblindWitness,
+    commitments_bundle: &ProofBundle,
+    reblind_bundle: &ProofBundle,
     global_state: &RelayerState,
     network_sender: &TokioSender<GossipOutbound>,
 ) -> Result<(), String> {
+    // Prove the link between the reblind and commitments proofs
+    let reblind_link_hint = &reblind_bundle.link_hint;
+    let comms_link_hint = &commitments_bundle.link_hint;
+    let linking_proof = link_sized_commitments_reblind(reblind_link_hint, comms_link_hint)
```

Comment:
perhaps obvious, but want to ask explicitly: computing link proofs is definitely lightweight enough that we don't need it to be separately handled in the proof manager?

Reply:
Correct